File: calibrate_data/calibrate_dataset.py
import os
import argparse
import math
import numpy as np
import ROOT
import matplotlib.pyplot as plt
from ROOT import gStyle, gPad, kRed

# load the RNO-G library
ROOT.gSystem.Load(os.environ.get('RNO_G_INSTALL_DIR')+"/lib/libmattak.so")

#voltage calibration 
bias_scan = "/data/condor_builds/users/avijai/RNO/tutorials-rnog/get_daqstatus/s13_r1794_scan.dat.gz"
data = "/data/i3store/users/rnog"

voltage_calib = ROOT.mattak.VoltageCalibration(bias_scan)
dataSet = ROOT.mattak.Dataset(13, 1796, voltage_calib, data)
dataSet_calib = dataSet.calibrated()

File: calibrate_data/calibrate_dataset.cpp
// C/C++ Includes
#include <iostream>
#include <stdio.h>

// ROOT Includes
#include "TTree.h"
#include "TFile.h"
#include "TCanvas.h"
#include "TH1D.h"

// mattak
#include "mattak/VoltageCalibration.h"
#include "mattak/Dataset.h"

int main(int argc, char **argv)
{

    std::string bias_scan = "/data/condor_builds/users/avijai/RNO/tutorials-rnog/get_daqstatus/s13_r1794_scan.dat.gz";
    std::string data_dir = "/data/i3store/users/rnog";

    mattak::VoltageCalibration vc = mattak::VoltageCalibration(bias_scan.c_str());
    mattak::Dataset d = mattak::Dataset(13, 1796, &vc, data_dir.c_str());
    auto cd = d.calibrated();

}
